packages/slice-machine/src/components/GenerateSliceWithAiModal/GenerateSliceWithAiModal.tsx

```diff
@@ -22,38 +27,84 @@ interface GenerateSliceWithAiModalProps {
 
 export function GenerateSliceWithAiModal(props: GenerateSliceWithAiModalProps) {
   const { open, onClose } = props;
+  const [slices, setSlices] = useState<Slice[]>([]);
 
   const onImagesSelected = (images: File[]) => {
-    console.log(images);
+    setSlices(
+      images.map((image, index) => ({
+        status: "loading",
+        displayName: `Slice ${index + 1}`,
+        image,
+      })),
+    );
   };
 
+  useEffect(() => {
+    if (slices.length > 0 && slices[0].status === "loading") {
+      const timeout = setTimeout(() => {
+        setSlices((slices) =>
+          slices.map((slice, index) => ({
+            ...slice,
+            displayName: `Hero ${index + 1}`,
+            thumbnailUrl:
+              "https://images.unsplash.com/photo-1588315029754-2dd089d39a1a?w=512",
+            status: "success",
+          })),
+        );
+      }, 2000);
+
+      return () => clearTimeout(timeout);
+    }
+  }, [slices]);
```

Comment:
use effect is not recommended for this, it's better to fire the action in the handler https://react.dev/learn/you-might-not-need-an-effect

Reply:
I removed `useEffect` and created a function to mock an API call. Let me know what you think.